core/pkg/fixhandler/yamlhandler.go

```diff
@@ -0,0 +1,286 @@
+		return nil, err
+	}
+	inputList := list.New()
+	var currentIndex uint
+
+	for {
+		candidateNode, errorReading := decoder.Decode()
+
+		if errors.Is(errorReading, io.EOF) {
+			switch reader := reader.(type) {
+			case *os.File:
+				safelyCloseFile(reader)
+			}
+			return inputList, nil
+		} else if errorReading != nil {
+			return nil, fmt.Errorf("bad file '%v': %w", filename, errorReading)
+		}
+		candidateNode.Document = currentIndex
+		candidateNode.Filename = filename
+		candidateNode.FileIndex = fileIndex
+		candidateNode.EvaluateTogether = true
+
+		inputList.PushBack(candidateNode)
+
+		currentIndex = currentIndex + 1
+	}
+}
+
+func safelyCloseFile(file *os.File) {
+	err := file.Close()
+	if err != nil {
+		fmt.Println("Error Closing File")
+	}
+}
+
+func getLineAndContentToAdd(node *yaml.Node) *[]LineAndContentToAdd {
+	contentToAdd := make([]LineAndContentToAdd, 0)
+	getLineAndContentToAddHelper(0, node, &contentToAdd)
+	return &contentToAdd
+}
+
+func getLineAndContentToAddHelper(nodeIdx int, parentNode *yaml.Node, contentToAdd *[]LineAndContentToAdd) {
+	node := parentNode.Content[nodeIdx]
+	var content string
+	var err error
+	if node.Line == 0 && node.Column == 0 {
+		if parentNode.Kind == yaml.MappingNode {
+			if nodeIdx%2 != 0 {
+				return
+			}
+		}
+		content, err = enocodeIntoYaml(parentNode, nodeIdx)
+
+		if err != nil {
+			fmt.Println("Cannot Encode into YAML")
+		}
+
+		indentationSpacesBeforeContent := parentNode.Column - 1
+
+		content = addIndentationToContent(content, indentationSpacesBeforeContent)
+
+		// Getting the line to add content after. Add directly after the left Sibling.
+		var lineToAddAfter int
+		for idx := nodeIdx - 1; idx >= 0; idx-- {
+			if parentNode.Content[idx].Line != 0 {
+				lineToAddAfter = getEndingLine(idx, parentNode)
+			}
+		}
+		lineAndContentToAdd := LineAndContentToAdd{
+			Line:    lineToAddAfter,
+			Content: content,
+		}
+		*contentToAdd = append(*contentToAdd, lineAndContentToAdd)
+	}
+
+	for index, _ := range node.Content {
+		getLineAndContentToAddHelper(index, node, contentToAdd)
+	}
+}
+
+func enocodeIntoYaml(parentNode *yaml.Node, nodeIdx int) (string, error) {
+	if parentNode.Kind == yaml.MappingNode {
+		content := make([]*yaml.Node, 0)
+		content = append(content, parentNode.Content[nodeIdx], parentNode.Content[nodeIdx+1])
+		parentForContent := yaml.Node{
+			Kind:    yaml.MappingNode,
+			Content: content,
+		}
+		buf := new(bytes.Buffer)
+		encoder := yaml.NewEncoder(buf)
+		errorEncoding := encoder.Encode(parentForContent)
+		if errorEncoding != nil {
+			return "", fmt.Errorf("Error debugging node, %v", errorEncoding.Error())
+		}
+		errorClosingEncoder := encoder.Close()
+		if errorClosingEncoder != nil {
+			return "", fmt.Errorf("Error closing encoder: ", errorClosingEncoder.Error())
+		}
+		return fmt.Sprintf(`%v`, buf.String()), nil
+	}
+
+	return "", nil
+}
+
+func addIndentationToContent(content string, indentationSpacesBeforeContent int) string {
```

Comment:
suggestion: we should name the function more concisely

We could rename it as `indentContent` to save on typed characters :)